pennylane/_device.py

```diff
@@ -302,33 +309,69 @@ def supports_observable(self, observable):
         if isinstance(observable, str):
             return observable in self.observables
 
-        raise ValueError("The given operation must either be a pennylane.Observable class or a string.")
+        raise ValueError("The given observable must either be a pennylane.Observable class or a string.")
+
+    def get_original_operation_name(self, operation_name: str):
```

Comment:
I would refrain from specifying types for arguments, just because we don't do it anywhere else in the codebase.

Reply:
This is fine - I don't think it's strictly necessary that type hinting be an 'all or nothing' approach in Python, as there are cases where we want to take advantage of duck-typing, and other cases where we want stricter type linting. Adding type hints where it makes sense can help lift the overall code quality :)

Reply:
Type hinting by using type annotations could be really useful to improve could readability. For people who have a background with statically typed languages, it provides a lot of help to have these at least in the signature of the methods and functions.

That being said, in this code types are (mostly) included both in type hints and in the docstring. I believe that although there is a certain redundancy to this approach, they can be used in different ways. On one hand type hints can be good for developing (if using an IDE that takes advantage of it) as well as can be utilized potentially for static analysis. Docstrings, on the other hand, are nice such that the documentation is more complete.